exporter/zipkinexporter/zipkin.go

```diff
@@ -175,6 +212,49 @@ func (ze *zipkinExporter) Shutdown() error {
 	return ze.reporter.Close()
 }
 
+func (ze *zipkinExporter) PushTraceData(ctx context.Context, td consumerdata.TraceData) (droppedSpans int, err error) {
+	tbatch := []*zipkinmodel.SpanModel{}
+	for _, span := range td.Spans {
+		sd, err := spandatatranslator.ProtoSpanToOCSpanData(span)
+		if err != nil {
+			// return len(td.Spans) - goodSpans, consumererror.Permanent(err)
+			return len(td.Spans), consumererror.Permanent(err)
+		}
+		zs := ze.zipkinSpan(td.Node, sd)
+		tbatch = append(tbatch, &zs)
+	}
+
+	body, err := ze.serializer.Serialize(tbatch)
+	if err != nil {
+		// r.logger.Printf("failed when marshalling the spans batch: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+
+	req, err := http.NewRequest("POST", ze.url, bytes.NewReader(body))
+	if err != nil {
+		// r.logger.Printf("failed when creating the request: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+	req.Header.Set("Content-Type", ze.serializer.ContentType())
+	/*
+		if r.reqCallback != nil {
+			r.reqCallback(req)
+		}*/
+
+	client := &http.Client{Timeout: defaultTimeout}
```

Comment:
I am not sure about we creating a new client on every report. I'd say it is better to hold a client when constructing the zipkinExporter.

Reply:
Sure, will do. Thanks @jcchavezs  and @pjanotti